pkg/mermaid/README.md

```diff
@@ -0,0 +1,47 @@
+
+**Integration Diagram** (https://github.com/anz-bank/sysl/pull/787)
+
+Integration diagram can be generated by three different interfaces.
+
+- GenerateFullIntegrationDiagram takes in the sysl module, iterates through all its applications and stores all the calls that each of the applications makes to other applications and generates and stores the appropriate mermaid code for that relationship.
+- GenerateIntegrationDiagram takes in an application and does the similar process, but finds all the relationships for that application only.
+- GenerateMultipleAppIntegrationDiagram takes in a list of applications and follows the same process.
+
+**Endpoint Analysis Diagram** (https://github.com/anz-bank/sysl/pull/803)
+
+Endpoint analysis diagram generator is an extension of integration diagram generator, but it adds the concept of subgraph from mermaid in order to display the endpoints of each of the applications in the generated diagram.
+
+**Data-model Diagram** (https://github.com/anz-bank/sysl/pull/838)
+
+Data-model diagram generator uses the sysl wrapper library to simplify the process of iterating through types defined in different applications. There are two interfaces to generating diagrams here.
+
+- GenerateFullDataDiagram takes a sysl module and uses the sysl wrapper library to generating a map of strings to Type. The types in these maps can be the either of the following – tuple, relation, enum, map or empty. Tuples and relations consist of primitive types, lists of other types or references to other types. If a type refers to or is a list of another type, this relationship is stored and printed in the diagram. Maps take a similar approach as well. Enums do not contain any references, thus appropriate mermaid code is printed directly.
+- GetDataDiagramWithAppAndType takes a sysl module, along with an app and a type. This runs in a similar fashion to the previous interface but it runs through it twice. The first iteration captures all the relationship that exists in that module and this result is used to find all the relationship that the given app and types has to other types. The resulting types are run through in the second iteration and the appropriate mermaid code is generated along with the relationships between the printed types. This uses the concept of class diagrams in mermaid.
+
+**Alternatives Considered**
+
+- The current approach in sequence, integration and endpoint analysis uses recursion which makes the resulting string non deterministic which makes it hard to assert the resulting mermaid code. Though it doesn't change the resulting diagram, it would be nice to have a map that hold all the information and iterate through it so that the resulting code is in order.
+
+**Enhancements/Future works**
+
+- Enhancement on this would be to add a logger to detect if there are any nil references for bad inputs.
+- Escape special characters instead of replacing them in mermaid code. (Issue on mermaid)
```

Comment:
Is there an issue that you can link to for any of these? Otherwise some details around the special characters which aren't currently supported by Mermaid would be super helpful here